app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = Post.where(author_id: @user.id)
+    @posts = Post.includes(:author).where(author_id: @user.id)
```

Comment:
- [ ] Great job on the n+1 problems but for posts to work as expected solving n+1 problems, Please update the code with the below one.

```ruby
@posts = Post.includes(:comments).where(author_id: @user.id)